sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -351,4 +352,28 @@ public static Duration getDefaultTimeoutFromEnvironment(Configuration configurat
             return defaultTimeout;
         }
     }
+
+    /**
+     * Merges two {@link Context Contexts} into a new {@link Context}.
+     *
+     * @param into Context being merged into.
+     * @param from Context being merged.
+     * @return A new Context that is the merged Contexts.
+     * @throws NullPointerException If either {@code into} or {@code from} is null.
+     */
+    public static Context mergeContexts(Context into, Context from) {
```

Comment:
Do we need this method to specifically be `from` and `into`. Can it be modified/generalized to merge the shorter linked list (context) to be the `from` here?

Reply:
Need to make sure I'm getting this correctly, are you asking if we can determine which of the two `Context`s is short and use that one as the one being merging into the longer one?

Reply:
yes, exactly. Like does it matter which is `from` and `to`?

Reply:
The ordering of the merging does matter, given the following values for two Contexts:

Context 1
```
key1 : value1
key2 : value2
key3 : value3
```

Context 2
```
key1 : value4
key3 : value5
```

If Context 2 is merged into Context 1 the outcome is:
```
key1 : value1
key2 : value2
key3 : value3
key1 : value4
key3 : value5
```

If Context 1 is merged into Context 2 the outcome is:
```
key1 : value4
key3 : value5
key1 : value1
key2 : value2
key3 : value3
```

Given that value look-ups in Context will stop on the first found (bottom-up) these two result in different runtime behaviors. So, if `key3` was looked up in the first scenario it would result in `value5` and in the second scenario it would return `value3`.

Reply:
Given the use case, and the ordering of the context data matters, we would need a defined `from` and `to`.